### PR TITLE
Add HTTP header checking to end-to-end integration tests

### PR DESCRIPTION
**Pull request by antekresic**
This commit adds additional tests which check that HTTP responses from
Prometheus and the connector contain matching HTTP headers.

### CODE REVIEW

**Review by JLockerman — CHANGES_REQUESTED** [review]
I don't think the `.bak` and `.stash` files should be included

File: pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go [hunk 1]
```diff
@@ -122,30 +126,35 @@ func testRequest(req *http.Request, handler http.Handler, client *http.Client, c
...
 		wg := sync.WaitGroup{}
 
 		for i := 0; i < len(perm); i++ {
-			req := requestCases[perm[i]].req
+			tsReq := requestCases[perm[i]].tsReq
+			promReq := requestCases[perm[i]].promReq
 			log := requestCases[perm[i]].log
 			wg.Add(1)
-			go func() {
+			go func(log string) {
```
Comment (cevian) [inline pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go:141]: why was this change necessary? Log is created inside the loop so it should be local to the closure

File: pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go [hunk 2]
```diff
@@ -177,3 +186,39 @@ func testRequestConcurrent(requestCases []requestCase, handler http.Handler, cli
 		wg.Wait()
 	}
 }
+
+func compareHTTPHeaders(t *testing.T, first, second http.Header) {
```
Comment (cevian) [inline pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go:190]:
```suggestion
func compareHTTPHeaders(t *testing.T, expected, actual http.Header) {
```

File: pkg/pgmodel/end_to_end_tests/promql_label_endpoint_test.go [hunk 3]
```diff
@@ -75,28 +77,39 @@ func TestPromQLLabelEndpoint(t *testing.T) {
...
+		}
 		labelNamesHandler := api.Labels(apiConfig, queryable)
 		labelValuesHandler := api.LabelValues(apiConfig, queryable)
 
 		router := route.New()
 		router.Get("/api/v1/label/:name/values", labelValuesHandler.ServeHTTP)
 		router.Get("/api/v1/labels", labelNamesHandler.ServeHTTP)
 
-		apiURL := fmt.Sprintf("http://%s:%d/api/v1", testhelpers.PromHost, testhelpers.PromPort.Int())
+		ts := httptest.NewServer(router)
```
Comment (cevian) [inline pkg/pgmodel/end_to_end_tests/promql_label_endpoint_test.go:90]: Is this the actual change that fixed the issue? If so I would explain in the commit message a bit more....

**Review by cevian — APPROVED** [review]
Overall I really like these changes. I have a hard time understanding what was wrong before and what's the actual fix here. I'd put more of a description in the commit message.

File: pkg/pgmodel/end_to_end_tests/promql_label_endpoint_test.go (hunk 3 above)
Comment (antekresic) [reply pkg/pgmodel/end_to_end_tests/promql_label_endpoint_test.go:90]: Yep, will update it.

File: pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go (hunk 1 above)
Comment (antekresic) [reply pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go:141]: I'll remove that, its leftover, not used anymore.

**Review by JLockerman — APPROVED** [review]
